elasticgraph-warehouse/spec/integration/elastic_graph/warehouse/schema_definition/rake_tasks_spec.rb

```diff
@@ -0,0 +1,219 @@
+# Copyright 2024 - 2025 Block, Inc.
+#
+# Use of this source code is governed by an MIT-style
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/MIT.
+#
+# frozen_string_literal: true
+
+require "elastic_graph/constants"
+require "elastic_graph/schema_definition/rake_tasks"
+require "elastic_graph/warehouse/schema_definition/api_extension"
+require "yaml"
+
+module ElasticGraph
+  module Warehouse
+    module SchemaDefinition
+      RSpec.describe "Warehouse RakeTasks", :rake_task, :in_temp_dir do
+        describe "schema_artifacts:dump" do
+          it "dumps warehouse artifact when warehouse tables are defined" do
+            write_warehouse_schema(table_defs: <<~EOS)
+              s.object_type "Product" do |t|
+                t.field "id", "ID"
+                t.field "name", "String"
+                t.field "price", "Float"
+                t.index "products" do |i|
+                  i.warehouse_table "products"
+                end
+              end
+            EOS
+
+            expect {
+              output = run_rake_with_warehouse("schema_artifacts:dump")
+              expect(output.lines).to include(
+                a_string_including("Dumped", Warehouse::DATA_WAREHOUSE_FILE)
+              )
+            }.to change { read_artifact(Warehouse::DATA_WAREHOUSE_FILE) }
+              .from(a_falsy_value)
```

Comment:
```suggestion
              .from(nil)
```

`nil` is a bit more precise (`a_falsy_value` encompasses both `nil` and `false`).

Edit: after playing around locally I see that actually `false` is being returned. But if you accept my suggested change to `read_artifact`, it'll return `nil` instead and then you can use `nil` here.